src/aegis_ai/tools/http_get/__init__.py

```diff
@@ -0,0 +1,158 @@
+# intentionally primitive HTTP GET retrieval tool
+
+import logging
+import os
+import re
+from urllib.parse import urlparse
+
+import requests
+
+from typing import Optional, Literal
+from pydantic import BaseModel, Field
+from pydantic_ai import Tool, RunContext
+
+from aegis_ai.tools import default_tool_http_headers
+
+logger = logging.getLogger(__name__)
+
+
+def compile_allowed_hosts_regex() -> Optional[re.Pattern]:
+    """
+    Loads AEGIS_HTTP_GET_TOOL_ALLOW_LIST from env, which is a comma-delimited list in a string.
+    Escapes non-regex characters where necessary, and compiles patterns into a single, case-insensitive
+    regex using OR operator (|).
+    """
```

Comment:
Honestly, this looks wild.  I think we should clearly mark this as an experimental feature.  If we ever want to support this as a security feature, we should go for a syntax that is already being used elsewhere, such as the `network.negotiate-auth.trusted-uris` configuration field in Firefox, or `AuthServerAllowlist` from Google Chrome (and ideally reuse some already trusted solution for this).  Otherwise this is going to be a frequent source of security issues.